br/pkg/lightning/importer/table_import.go

```diff
@@ -1176,3 +1204,285 @@ func (tr *TableImporter) analyzeTable(ctx context.Context, g glue.SQLExecutor) e
 	task.End(zap.ErrorLevel, err)
 	return err
 }
+
+func (tr *TableImporter) addIndexes(ctx context.Context, db *sql.DB) (retErr error) {
+	task := tr.logger.Begin(zap.InfoLevel, "add indexes")
+	defer func() {
+		task.End(zap.ErrorLevel, retErr)
+	}()
+
+	tblInfo := tr.tableInfo
+	tableName := common.UniqueTable(tblInfo.DB, tblInfo.Name)
+
+	batchSQL, sqls := buildAddIndexSQL(tableName, tblInfo.Core, tblInfo.Desired)
+	if len(sqls) == 0 {
+		return nil
+	}
+
+	err := tr.executeDDL(ctx, db, batchSQL)
+	if err == nil || !shouldIgnoreDDLError(err) || len(sqls) == 1 {
```

Comment:
when `err can be ignored` and `len(sqls) == 1`, we return err?

Reply:
We will continue adding indexes one by one when encountering errors like `ErrDupUnique`. `shouldIgnoreDDLError` is not a good name, I will improve it later.